File: backend/src/RealEstate.Application/DTOs/Leads/LeadDetailsDto.cs
using RealEstate.Domain.Enums.Leads;

namespace RealEstate.Application.DTOs.Leads;

public sealed record LeadDetailsDto(
    Guid Id,
    Guid PropertyId,
    string FullName,
    string? Email,
    string? PhoneNumber,
    string? Message,
    LeadStatus Status,
    DateTime CreatedAt,
    DateTime? UpdatedAt
);

File: frontend/src/shared/api/dtos/leads/lead-details.dto.ts
import type {LeadStatus} from "@/shared/types/leads.ts";

export type LeadDetailsDto = {
  id: string;
  propertyId: string;
  fullName: string;
  email?: string;
  phoneNumber?: string;
  message?: string;
  status: LeadStatus;
  createdAt: Date;
  updatedAt: Date;
};
